### ISSUE TITLE
[med] Color Contrast: Many Pages: Input controls grey border against adjacent white fails contrast

### ISSUE THREAD

**Issue by ogumen** [Expensify/App#5521]
## Action Performed:
Prerequisite: user has created personal account and is logged into same.
1. Using Chrome, open https://staging.new.expensify.com/
2. Click on the new chat floating action icon button (+) in the left side panel and select 'New Group'
3. In the 'New Group' side-panel overlay, inspect the contrast ratio of the light grey border around the search input field against the adjacent white background.

## Expected Result:
The graphical information required to identify user interface components must have a contrast ratio of at least 3.00:1 against adjacent colors. In this case, the input field is visually identified by a light grey border that shows its position, size and lets users know where to click, tap, etc. The border color must have a contrast ratio greater that 3.00:1 against the adjacent white background.

## Actual Result:
The light grey border (#ECECEC) against the adjacent white background (#FFFFFF) has a contrast ratio of 1.18:1, rendering it almost invisible (1.0:1).

HTML:
`<input placeholder="Name, email, or phone number" autocapitalize="sentences" autocomplete="on" autocorrect="on" dir="auto" spellcheck="true" type="text" class="css-11aywtz r-tz77pi" value="" style="background-color: rgb(255, 255, 255);border-color: #ececec;border-radius: 8px;border-width: 1px;color: rgb(11, 27, 52);font-family: GTAmericaExp-Regular;font-size: 15px;height: 42px;padding: 10px 12px;vertical-align: middle;">`

CSS:
`element.style {
background-color: rgb(255, 255, 255);
border-color: #ececec;
...
}`

Other occurrences:
1. The grey circle for custom checkboxes adjacent to contact names in the panel has the same color contrast ratio.
2. Chat:
- Search panel search field
- Emoji panel search field
- Chat Message field

## Workaround:
No workaround

## Area issue was found in:
Create new group overlay, Chat page

## Failed WCAG checkpoints
1.4.11

## User impact:
Many colorblind and other sight impaired users will not identify the area as a control and may not know where to click.

## Suggested resolution:
Use a darker shade of grey for the border color, such as `#919191`, which will give a color contrast ratio of 3.15:1.

## Platform:
Where is this issue occurring?

- Web

**Version Number:**
**Reproducible in staging?:**Yes
**Reproducible in production?:**
**Notes/Photos/Videos:** Any additional supporting documentation
**Issue reported by:**
![Bug5183162_Color_Contrast-Chat-Input_field_grey_border_against_adjacent_white_fails_contrast-Emoji](https://user-images.githubusercontent.com/88733897/134738792-e8dae56e-9055-4b34-8a90-b3dc20204fa2.png)
![Bug5183162_Color_Contrast-Chat-Input_field_grey_border_against_adjacent_white_fails_contrast-Message_entry](https://user-images.githubusercontent.com/88733897/134738797-57b6c7c9-f54a-4b84-b8f1-5ece51d7d7a6.png)
![Bug5183162_Color_Contrast-Chat-Input_field_grey_border_against_adjacent_white_fails_contrast-Search](https://user-images.githubusercontent.com/88733897/134738802-45da2794-bd2f-4d86-96f8-6c140615a3e1.png)
![Bug5183162_Color_Contrast-Create_New_Group-Custom_Checkbox_grey_border_against_adjacent_white_fails_contrast](https://user-images.githubusercontent.com/88733897/134738805-be357719-c605-4456-b6fb-0967e85a2f02.png)
![Bug5183162_Color_Contrast-Create_New_Group-Input_field_grey_border_against_adjacent_white_fails_contrast](https://user-images.githubusercontent.com/88733897/134738808-a841a8e6-c3de-43d7-b3e8-d2ae24e91430.png)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by shawnborton** [comment]
Let's also add checkboxes to the list of inputs to update:
![image](https://user-images.githubusercontent.com/2319350/134877628-50932351-9b4a-4153-9fee-81027197de5f.png)

**Comment by shawnborton** [comment]
@ogumen what is the minimum contrast ratio that we need here?

**Comment by flodnv** [comment]
This should be as easy as just changing the border color to a new one, let's see if someone external will do this.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @kadiealexander (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by ogumen** [comment]
> 
> 
> @ogumen what is the minimum contrast ratio that we need here?

@shawnborton The minimum contrast ratio of the UI components should be at least 3.00:1.

**Comment by PrashantMangukiya** [comment]
#### Proposed Solution

I did brief research on how border color applied to ui element. i.e. We have  `border: colors.gray2` within 
 src/styles/themes/default.js file that suppose to use for border to any ui element.

So we expect to change border: `#919191` i.e. dark color. But as soon as we change border color it have side effect many places within ui because border is used as selected list color also as shown below screenshot.
<img width="1093" alt="Screenshot 2021-09-27 at 7 45 24 PM" src="https://user-images.githubusercontent.com/7823358/134928480-77c20cdb-545e-4e1b-8dde-f1742fad0e39.png">

As we need dark border for some input component. So we can go with this approach also. i.e. within src/styles/colors.js add one more color e.g. `gray5: '#919191'` and within src/styles/themes/default.js file add  `borderDark: colors.gray5` and thereafter we can apply borderDark to ui input component we need. 

Other way is to update grey color to `border: colors.gray2` --> `border: colors.gray5`  and correct correct color from selected list etc. where border is used. Not sure at present how many places border is used that suppose to not.

I think we can discuss and figure out which way is better to go with.

**Comment by shawnborton** [comment]
Yeah I think the solution is as simple as just changing out the border color we use to a darker shade. That being said, I do think @Expensify/design should take a look at these accessibility issues altogether and chat about them quickly internally to make sure we are finding the right new colors to use in these situations. Going to assign to myself for now.

**Comment by shawnborton** [comment]
[Started a conversation](https://expensify.slack.com/archives/CNKBZSQ07/p1632833126029100) over in #accessibility about this

**Comment by shawnborton** [comment]
Okay so what I would recommend doing here... in `colors.js` we can modify our grays a bit:

```
    gray4: '#7D8B8F',
```

change that to

```
    gray4: '#8B979A',
    gray5: '#6B7880',
```

Then from there, you will want to make sure the `border` color in `themes/default.js` uses `gray4`.

Bonus points if you want to update the [supporting text/label color issue](https://github.com/Expensify/App/issues/5503), which you could just update `textSupporting` to `gray5`

Bonus points if you want to tackle the [icon color issue](https://github.com/Expensify/App/issues/5506) here too, which you could just update `icon` to use `gray4` and it will pass contrast tests. 

Let me know how all of that sounds!

**Comment by MelvinBot** [comment]
Triggered auto assignment to @stephanieelliott (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by Beamanator** [comment]
Adding `External` again since it looks like we're opening up to proposals again!

**Comment by PrashantMangukiya** [comment]
Hi @shawnborton, I did quick test. i.e. within `colors.js` set `gray4: '#8B979A'` and  within `themes/default.js` set `border: colors.gray4`  It looks like that border used in many components. e.g. within LHN also border color used as selected record.  

I think we need to decide strategy how to approach this, because if we set `border: colors.gray4` then it will affect many component look and feel as shown below screenshot.

<img width="1118" alt="Screenshot 2021-09-30 at 6 23 08 PM" src="https://user-images.githubusercontent.com/7823358/135465165-2dc6892c-ed8d-4b44-a419-24d05d9479b7.png">

At present trying to figure out best approach, finger crossed.

**Comment by shawnborton** [comment]
Good points in the above. 

That being said, the hover style for the chat row seems wrong - that should not be using the same color as borders, we have a specific variable for that: `activeComponentBG: colors.gray2,`

I think what I would recommend then is that we have two separate border variables in `themes/default.js`:
- `componentBorder` - this is the border used for all UI components like inputs, etc. It uses `gray4`
- `accentBorder` - this is the border used for things like the right edge of the chats list, the bottom border on the chat header area, as well as the small bar next to the `+` in the compose box.

**Comment by PrashantMangukiya** [comment]
Temporary I set `border: '#ff0000'`  i.e. red within `themes/default.js` to get the idea how many components using the border props from default.

Below are some screenshot that shows usage of `border` props from `default.js`,  (this is not all list) but gives idea how many components need to update border props etc. Just sharing so it will help us during discussion.

<img width="1101" alt="Screenshot 2021-09-30 at 8 38 17 PM" src="https://user-images.githubusercontent.com/7823358/135486687-3f8f8035-830e-4081-bcd4-454ab575a5b7.png">

<img width="1101" alt="Screenshot 2021-09-30 at 8 38 24 PM" src="https://user-images.githubusercontent.com/7823358/135486751-ecc8e937-9254-4766-9b38-ef47fcbc3ac1.png">

<img width="1101" alt="Screenshot 2021-09-30 at 8 38 44 PM" src="https://user-images.githubusercontent.com/7823358/135486806-81936c74-d737-4fd6-ab87-065a8bb1f753.png">

<img width="1101" alt="Screenshot 2021-09-30 at 8 38 54 PM" src="https://user-images.githubusercontent.com/7823358/135486881-f7cba48d-2d7e-4ca9-98b9-5f8223d29a1b.png">

<img width="1101" alt="Screenshot 2021-09-30 at 8 39 37 PM" src="https://user-images.githubusercontent.com/7823358/135486941-1e3939d1-847d-4429-b905-cd8f363265b7.png">

**Comment by shawnborton** [comment]
Thanks for doing that. I still think my [proposed solution](https://github.com/Expensify/App/issues/5521#issuecomment-931401162) to have two border colors (one for components, one for accents) would be great here.

**Comment by Beamanator** [comment]
Still waiting for proposals based on Shawn's latest comments

**Comment by Beamanator** [comment]
@PrashantMangukiya are you still interested in working on this issue? If so, can you submit an updated proposal?

**Comment by mateomorris** [comment]
# Proposal 

* In `colors.js`: Change `gray4` to `#8B979A` and add `gray5: '#6B7880'

In `default.js`: 
* Remove `border`
* Add `componentBorder: colors.gray4` and `accentBorder: colors.gray2`
* Change `textSupporting: colors.gray4` to `textSupporting:  colors.gray5` #5503
* Change `icon: colors.gray3` to `icon: colors.gray4` #5506

In `styles.js`: update `sidebarLinkActive.backgroundColor` from `backgroundColor: themeColors.border` to `backgroundColor: themeColors.activeComponentBG`

In `styles.js`, `getReportActionContextMenuStyles.js`, and `getModalStyles`, change every instance of `themeColors.border` to `themeColors.accentBorder`

Then we can selectively change all the inputs to use `themeColors.componentBorder` as needed (still hunting those down).

**Comment by stephanieelliott** [comment]
This is on Upwork here: https://www.upwork.com/jobs/~019b45fec6e1a6cdcc

**Comment by Beamanator** [comment]
Sounds good, thanks for the heads up @parasharrajat 💪

**Comment by stephanieelliott** [comment]
Hired on Upwork, thanks @mateomorris!

**Comment by mateomorris** [comment]
Thanks @stephanieelliott! Accepted

**Comment by stephanieelliott** [comment]
All set and paid in Upwork. Thanks @mateomorris!
